### PR TITLE
[payment] Created all payment functionality

### PR DESCRIPTION
**Pull request by Skrudj**

### CODE REVIEW

**Review by ArtemGrunin — APPROVED** [review]

File: src/main/java/com/example/carsharingservice/model/Payment.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+enum Status {
+    PENDING,
+    PAID
+}
+
+enum Type {
+    PAYMENT,
+    FINE
+}
```
Comment (yurihhhT) [inline src/main/java/com/example/carsharingservice/model/Payment.java:34]: Enum should be inside class

File: src/main/java/com/example/carsharingservice/model/Payment.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+@Entity
+@Table(name = "payments")
+public class Payment {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    int sessionId;
+    int rentalId;
+    URL url;
+    BigDecimal paymentAmount;
+    Type type;
```
Comment (yurihhhT) [inline src/main/java/com/example/carsharingservice/model/Payment.java:22]: Better to use Integer or Long for ID
Also better to use private modification to each field

**Review by yurihhhT — CHANGES_REQUESTED** [review]

**Review by ArtemPonomar — APPROVED** [review]

File: src/main/java/com/example/carsharingservice/repository/PaymentRepository.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.example.carsharingservice.repository;
+
+import com.example.carsharingservice.model.Payment;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface PaymentRepository extends JpaRepository<Payment, Long> {
+    Payment update(Payment payment);
```
Comment (mishok321) [inline src/main/java/com/example/carsharingservice/repository/PaymentRepository.java:9]: redunant

File: src/main/java/com/example/carsharingservice/model/Payment.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+@Getter
+@Setter
+@Entity
+@Table(name = "payments")
+@SQLDelete(sql = "UPDATE payments SET isDeleted = true WHERE id=?")
+@Where(clause = "isDeleted=false")
+public class Payment {
+    @Id
+    @GeneratedValue()
+    private Long rentalId;
```
Comment (mishok321) [inline src/main/java/com/example/carsharingservice/model/Payment.java:25]:
```suggestion
    private Long id;

```

and add foreign key

**Review by mishok321 — CHANGES_REQUESTED** [review]

**Review by mishok321 — APPROVED** [review]

**Review by yurihhhT — APPROVED** [review]

**Review by yurihhhT — APPROVED** [review]

**Review by yurihhhT — APPROVED** [review]

**Review by yurihhhT — APPROVED** [review]

**Review by yurihhhT — APPROVED** [review]

**Review by yurihhhT — APPROVED** [review]
1

**Pull request merged**